ctk/common/src/main/kotlin/org/codice/compliance/verification/core/assertions/StatementVerifier.kt

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.verification.core.assertions
+
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_2_7_2
+import org.codice.compliance.SAMLCore_2_7_3
+import org.codice.compliance.SAMLCore_2_7_3_2_a
+import org.codice.compliance.SAMLCore_2_7_4_a
+import org.codice.compliance.allChildren
+import org.codice.compliance.children
+import org.codice.compliance.utils.TestCommon.Companion.ELEMENT
+import org.codice.compliance.verification.core.CommonDataTypeVerifier
+import org.w3c.dom.Node
+
+internal class StatementVerifier(val node: Node) {
+    companion object {
+        private const val TYPE = "Type"
+        private const val SUBJECT = "Subject"
+        private const val ASSERTION = "Assertion"
+    }
+
+    /** 2.7 Statements */
+    fun verify() {
+        verifyAuthnStatement()
+        verifySubjectLocality()
+        verifyAuthnContext()
+        verifyAttributeStatement()
+        verifyAttribute()
+        verifyEncryptedAttribute()
+        verifyAuthzDecisionStatement()
+        verifyAction()
+    }
+
+    /** 2.7.2 Element <AuthnStatement> **/
+    private fun verifyAuthnStatement() {
+        node.allChildren("AuthnStatement").forEach {
+            val sessionIndex = it.attributes.getNamedItem("SessionIndex")
+            val sessionNotOnOrAfter = it.attributes.getNamedItem("SessionNotOnOrAfter")
+
+            CommonDataTypeVerifier.verifyDateTimeValues(it.attributes.getNamedItem("AuthnInstant"))
```

Comment:
✏️ Could move this line to be the first in the closure so it can short-circuit finding the index and notOnOrAfter values.